node/src/main/scala/coop/rchain/node/api/WebApi.scala

```diff
@@ -97,11 +101,27 @@ object WebApi {
         .flatMap(BlockAPI.deploy(_, triggerProposeF, minPhloPrice, isNodeReadOnly))
         .flatMap(_.liftToBlockApiErr)
 
-    def listenForDataAtName(req: DataRequest): F[DataResponse] =
+    def listenForDataAtName(req: DataAtNameRequest): F[DataAtNameResponse] =
       BlockAPI
         .getListeningNameDataResponse(req.depth, toPar(req), apiMaxBlocksLimit)
         .flatMap(_.liftToBlockApiErr)
-        .map(toDataResponse)
+        .map(toDataAtNameResponse)
+
+    def getDataAtPar(req: DataAtParRequest): F[RhoDataResponse] = {
+      implicit val rm: RuntimeManager[F] = runtimeManager
+
+      val deployId = req.name match {
+        case unforg: UnforgPrivate  => unforg.data
+        case unforg: UnforgDeploy   => unforg.data
+        case unforg: UnforgDeployer => unforg.data
+      }
```

Comment:
`UnforgDeploy` is a name (channel) that holds deploy signature and it's a special name different from `UnforgDeploy` and `UnforgPrivate`.
Here we are searching data that is stored on each of this names, so they must be supplied to `getDataAtPar` as received from WebAPI. BlockAPI should return the block info in the same way as `BlockAPI.getListeningNameDataResponse`.

Reply:
Resolved in [commit](https://github.com/rchain/rchain/pull/3584/commits/e06ff930cb4847f5b07ad3dbf54928028a3881db)